### ISSUE TITLE
Modifier key stays pressed

### ISSUE THREAD

**Issue by shungo27** [rvaiya/keyd#145]
Is there a way to solve this problem?

Expected key operation meta+c
⭕ + meta + shift - shift + c -> meta+c
❌ + shift + meta - shift + c -> shift+meta+c

Expected key operation alt+left
⭕ + alt + shift - shift + left -> alt+left
❌ + shift + alt - shift + left -> shift+alt+left

Expected key operation shift+left
⭕ + shift + alt + left - alt + left -> shift+left
❌ + alt + shift + left - alt + left -> shift+alt+left

\+ press / - release

default.conf

```
[ids]
*

[main]
leftmeta = overload(control, muhenkan)
rightmeta = overload(control, henkan)

[shift]
meta = layer(ctrl_shift)
alt = layer(alt_shift)

[control]
left = home
shift = layer(ctrl_shift)

[ctrl_shift:C-S]
left = S-home

[alt]
left = C-left
shift = layer(alt_shift)

[alt_shift:A-S]
left = S-C-left
```

keyd 2.2.6-beta

**Comment by rvaiya** [comment]
Unfortunately this is just a consequence of the way that layers work. In some circumstances this is desirable, but in the case of modifiers it presents the subtle issue you have described above. I have considered changing the default behaviour or adding a feature to address this, but I haven't encountered any issues in my day to day use or heard enough complaints to warrant the effort. Is this a major issue for you?

**Comment by shungo27** [comment]
Great! This issue has been fixed in v2.3.0-rc 😁

2.3.0-rc default.conf

```
[ids]
*

[main]
leftmeta = overload(control, muhenkan)
rightmeta = overload(control, henkan)

[shift]

[control]
left = home

[alt]
left = C-left

[ctrl+shift]
left = S-home

[alt+shift]
left = S-C-left
```

**Comment by rvaiya** [comment]
Indeed :). I was going to update this thread. The config you posted has a typo (`ctrl+shift` should be `control+shift` (unless you've defined `ctrl` somewhere else) but I assume you've already figured that out :P.
